### PR TITLE
SearchKit - Expose SavedSearch descriptions to admins

### PR DESCRIPTION
**Pull request by colemanw**
Overview
----------------------------------------
This allows SearchKit searches to have a description for admin use.
See https://lab.civicrm.org/dev/core/-/issues/3980

Before
----------------------------------------
Description field actually does exist in the db, but isn't shown.

After
----------------------------------------
image

image

Comments
----------------------------------------
@eileenmcnaughton added the field a while ago and I think I hesitated to add it to the UI because of clutter, but people are saying it would be useful.

### CODE REVIEW

File: CRM/Contact/DAO/SavedSearch.php [hunk 1]
```diff
@@ -320,6 +322,10 @@ public static function &fields() {
           'entity' => 'SavedSearch',
           'bao' => 'CRM_Contact_BAO_SavedSearch',
           'localizable' => 0,
+          'html' => [
```
Comment (larssandergreen) [inline CRM/Contact/DAO/SavedSearch.php:325]: Maybe just for my own understanding, but what's this doing?

Comment (colemanw) [reply CRM/Contact/DAO/SavedSearch.php:325]: It's not directly related to exposing "description" but while I was in there I wanted to clean up the metadata a bit. Adding the label in metadata means SearchKit can then automatically pick it up (and I was able to remove it below and just pass `true` to get that label in the table).

**Comment by larssandergreen** [comment]
Working well in general.

Not sure this should be only visible to admins. I think anyone who can see the search should be able to see the description.

For styling, it's getting a bit funky, here's my local (note left margin from the li and not full width):

and here's the test server (overhanging into next column):

I think the li is messing it up, maybe it would be happier just in a div like the label above?

Also, it's not getting the Bootstrap rounded corners, so I think the textarea is missing a class.

I think the edit in place on the list of searches should be a textarea rather than text. Shouldn't this be automagic? Otherwise, it's pretty hard to use.

File: CRM/Contact/DAO/SavedSearch.php (hunk 1 above)
Comment (larssandergreen) [reply CRM/Contact/DAO/SavedSearch.php:325]: Makes sense, I thought it was probably unrelated, thanks.

**Comment by larssandergreen** [comment]
Actually, the bootstrap rounded corners thing is probably just caching on my end.

**Comment by eileenmcnaughton** [comment]
The issues discussed appear unrelated

**Pull request merged**